marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -70,32 +121,22 @@ derive newtype instance monoidState :: Monoid State
 _holes :: Lens' State Holes
 _holes = _Newtype <<< prop (SProxy :: SProxy "holes")
 
-_negativePayments :: Lens' State (Array Position)
-_negativePayments = _Newtype <<< prop (SProxy :: SProxy "negativePayments")
-
-_negativeDeposits :: Lens' State (Array Position)
-_negativeDeposits = _Newtype <<< prop (SProxy :: SProxy "negativeDeposits")
-
 _maxTimeout :: Lens' State Timeout
 _maxTimeout = _Newtype <<< prop (SProxy :: SProxy "maxTimeout") <<< _Newtype
 
-_timeoutNotIncreasing :: Lens' State (Set Position)
-_timeoutNotIncreasing = _Newtype <<< prop (SProxy :: SProxy "timeoutNotIncreasing")
+_warnings :: Lens' State (Set Warning)
+_warnings = _Newtype <<< prop (SProxy :: SProxy "warnings")
 
 _letBindings :: Lens' State (Set ValueId)
 _letBindings = _Newtype <<< prop (SProxy :: SProxy "letBindings")
 
-_uninitializedUse :: Lens' State (Array Position)
-_uninitializedUse = _Newtype <<< prop (SProxy :: SProxy "uninitializedUse")
-
-_shadowedLet :: Lens' State (Array Position)
-_shadowedLet = _Newtype <<< prop (SProxy :: SProxy "shadowedLet")
-
-_trueObservation :: Lens' State (Array Position)
-_trueObservation = _Newtype <<< prop (SProxy :: SProxy "trueObservation")
-
-_falseObservation :: Lens' State (Array Position)
-_falseObservation = _Newtype <<< prop (SProxy :: SProxy "falseObservation")
+termToRange :: forall a. Show a => a -> { row :: Int, column :: Int } -> IRange
+termToRange a { row, column } =
+  { startLineNumber: row
+  , startColumn: column
+  , endLineNumber: row
+  , endColumn: column + (length (show a))
```

Comment:
We are going to need to change this at some point, because it doesn't work for multi-line warnings. But it requires changes to the parser.